lib/spack/spack/cmd/uninstall.py

```diff
@@ -44,10 +49,81 @@ def setup_parser(subparser):
     subparser.add_argument(
         '-a', '--all', action='store_true', dest='all',
         help="USE CAREFULLY. Remove ALL installed packages that match each " +
-        "supplied spec. i.e., if you say uninstall libelf, ALL versions of " +
-        "libelf are uninstalled. This is both useful and dangerous, like rm -r.")
+             "supplied spec. i.e., if you say uninstall libelf, ALL versions of " +
+             "libelf are uninstalled. This is both useful and dangerous, like rm -r.")
     subparser.add_argument(
-        'packages', nargs=argparse.REMAINDER, help="specs of packages to uninstall")
+        '-r', '--recursive', action='store_true', dest='recursive',
+        help='Uninstall all the packages that depends on the ones for which we required explicit removal.'
```

Comment:
I would reword this as "Also uninstall any packages that depend on this package."